### ISSUE TITLE
[Toast] Use Enum for duration

### ISSUE THREAD

**Issue by solonaarmstrong** [Shopify/polaris-react#1055]
# Feature request summary

@mateus had the idea to use an enum for the duration prop.

```
export enum ToastDuration {
  fast: 3000
  base: 5000 (this is the default)
  slow: 10000
  slowest: 15000
}
```

`duration?: ToastDuration | number`

## Rationale

Web reuses the same values: `duration={3000}` or `duration={5000}`. This would also work well with the action prop, when using at least 10,000ms.

---

🌟 Feature requests that are not yet planned will be closed. We then use the issue’s [:+1: upvotes](https://help.github.com/articles/about-conversations-on-github/) to track and set priorities. See the [contribution guidelines](https://github.com/Shopify/polaris-react/blob/master/.github/CONTRIBUTING.md) for more information.

**Comment by solonaarmstrong** [comment]
@dleroux @ry5n Do you have any thoughts?

**Comment by ry5n** [comment]
Is this a global set of values, or just for Toast?

**Comment by solonaarmstrong** [comment]
@ry5n Just for Toast.

**Comment by ry5n** [comment]
If it’s just for toast this could be helpful! I think it’s smart to add only the values we know are common though.

**Comment by dleroux** [comment]
Thats works. The current default is 3000 I believe so we might want to leave it as that. It might also help to provide a bit of guidance on the possible usage of each value so that there’s a bit of consistency?

**Comment by solonaarmstrong** [comment]
The default is 5000, but web uses 3000 in some places, so that value should be available. Yes, it would make sense to add a guidance around the values, too.

**Comment by solonaarmstrong** [comment]
https://github.com/Shopify/polaris-react/pull/1059

**Comment by AndrewMusgrave** [comment]
`slowest: 15000` seems extremely long to me, do we have any use cases for this? Maybe we should use `3000, 5000 , 7000, 9000`? How do we feel about using `base` as a value? We use it a lot in our sass functions as a default but we don't normally use it as a value e.g `spacing()` rather than `spacing(base)`.

**Comment by solonaarmstrong** [comment]
@AndrewMusgrave the recommended duration for a toast message with an action is at least 10000ms, to give merchants a chance to take the action, so that value should be in there. https://github.com/Shopify/polaris-react/pull/919/files#diff-f6f39e05b67f0498645df16bf2a7c4eaR128

I don't have strong feelings about 15000. I put it in there simply because our docs say **at least** 10000, which implies longer is good too.

I'm confused by your comment about base. There is a base value and it's the default of 5000.

The PR is here: https://github.com/Shopify/polaris-react/pull/1059 I didn't merge it yet to get more feedback on the values I chose, so please weigh in...

**Comment by ry5n** [comment]
@mateus Is this something you need? The associated PR has been closed.

**Comment by mateus** [comment]
> @mateus Is this something you need? The associated PR has been closed. If it’s something you need, please re-open. Thx!

Yeah, I don't think we felt the need for it as much as we thought we would. We can keep it closed and I can revisit in the future if needed. Thanks 👍
